skins/laika/src/components/pages/membership_form/MembershipType.vue

```diff
@@ -46,6 +49,22 @@ export default Vue.extend( {
 				return MembershipTypeModel;
 			},
 		},
+		AddressTypeModel: {
+			get: function () {
+				return AddressTypeModel;
+			},
+		},
+		...mapGetters( NS_MEMBERSHIP_ADDRESS, [ 'addressType' ] ),
+		isActiveTypeDisabled: {
+			get: function () {
+				if ( this.$store.getters[ NS_MEMBERSHIP_ADDRESS + '/addressType' ] === AddressTypeModel.COMPANY ) {
+					this.$data.selectedType = MembershipTypeModel.SUSTAINING;
+					this.setType();
```

Comment:
Please don't do a computed property with side effects. At least in my understanding of Vue, computed properties should only return values. Also, they are cached, so we can only kind-of rely on the change.

A better (or at least more explicit) approach IMHO would be to create a `mounted` handler where you add a watcher to the store. See https://dev.to/viniciuskneves/watch-for-vuex-state-changes-2mgj#vuex-watch for a code example and explanation

Reply:
Ok I see how having this additional logic in the getter does not look nice, but I don't think computed properties are meant to only return values. We can have a setter as part of a computed property where you'd make some changes based on what you called "side effects" and that will affect the getter. I committed a new version, where the logic is split between a getter and a setter. 
I thought a watcher's main usage is when you have async logic and we don't really have that here. IMHO it seems a bit too complex of a solution for the case that we have. It's *just* making an option disabled based on the value of another option.
So how about the setter solution?

Reply:
The problem with the setter is that it's not called anywhere (e.g. `isActiveTypeDisabled = false` in the template or `this.$props.isActiveTypeDisabled = false` in the code). But it can't be triggered by user interaction in this component, only by store changes, so using a setter here doesn't work.

The functionality we want to have (and test) is "When the address type in the store changes to 'company' and the selected type is `active`, then unselect the selected type." (or alternatively, depending on UX feedback  select "sustaining", although that would subvert the user choice, possibly out of the view of the user and thus not validated). 

> I thought a watcher's main usage is when you have async logic and we don't really have that here.

Watchers (in this case store watchers) are IMHO a good way to tackle the "when a store value changes, change `this.$data?" scenario in an event-based way. The article I linked to is the first result when I google for "vue change data when vuex changes"

>  I don't think computed properties are meant to only return values.

We're not using ESLint rules for Vue files, but among the plugin rules is one that says that computed properties should not have side effects: https://eslint.vuejs.org/rules/no-side-effects-in-computed-properties.html 
I take that as an indicator that computed properties should be pure functions (i.e. functions without side effects).

If we did not use a Buefy radio component, we could do away with the local state altogether and use a computed property and events, but unfortunately that pattern does not work well with the Buefy radio component which insists on having a v-model.